docs/containers-storage-architecture.md

```diff
@@ -0,0 +1,76 @@
+## containers-storage 1 "May 2024"
+
+## NAME
+containers-storage-architecture - The implementation of containers-storage 
+
+NOTE: These are implementation details subject to change. This document
+is intended to be used only as an aid for debugging and development
+of the containers-storage system. Do *not* directly access the filesystem
+components below from external code, only from the Go API.
+
+## DESCRIPTION
+The *containers/storage* library manages the following object types:
+
+- layer: A filesystem tree, along with metadata
+- image: A reference to a layer, along with metadata
+- container: A layer whose parent is an image
```

Comment:
These feel a bit too reductionist to me.

The important thing about images is that, unlike, layers, users frequently with deal with them directly; that they have human-usable names; and that they are nevertheless deduplicated.

The important thing about containers is… ? That the top layer is read-write? That a lot of the metadata is actually not in c/storage but in Podman’s / CRI-O’s data? Also, technically the parent of the container’s layer is still a layer.